refinery/tool_manager/schemas/ToolDefinition.json

```diff
@@ -1,45 +1,86 @@
 {
   "$schema": "http://json-schema.org/draft-04/schema",
-  "type": "object",
-
-  "required": [
-    "name",
-    "description",
-    "tool_type",
-    "file_relationship"
-  ],
-  "properties": {
-    "name": {
-      "type": "string"
-    },
-    "description": {
-      "type": "string"
-    },
-    "tool_type": {
-      "type": "string",
-      "enum": [
-        "WORKFLOW",
-        "VISUALIZATION"
-      ]
-    },
-    "file_relationship": {
-      "$ref": "FileRelationship.json#"
-    },
-    "output_files": {
-      "type": "array",
-      "items": {
-       "$ref": "OutputFile.json#"
-      },
-      "minItems": 1
+  "oneOf":[
```

Comment:
It looks like the two definitions are mostly the same: Did you look at using `allOf` to extend definitions? https://spacetelescope.github.io/understanding-json-schema/structuring.html#extending